common/script/ops/scoreTask.js

```diff
@@ -245,14 +241,10 @@ module.exports = function scoreTask (options = {}, req = {}) {
     }
   } else if (task.type === 'reward') {
     // Don't adjust values for rewards
-    delta += _changeTaskValue(user, task, direction, times, cron);
+    // If they're trying to purchase a too-expensive reward, don't allow them to do that.
+    if (task.value > user.stats.gp) throw new NotAuthorized(i18n.t('messageNotEnoughGold', req.language));
     // purchase item
     stats.gp -= Math.abs(task.value);
-    // hp - gp difference
-    if (stats.gp < 0) {
```

Comment:
Definitely. It hasn't actually been a feature for a long time because code further up in the file stops execution when the user doesn't have enough gold to buy a Reward, and if we reintroduced this feature, we'd get confused users complaining about their health decreasing when Rewards were purchased. It's not a good idea any more. That block is just ancient, unused code that should have been cleaned up years ago. :)